chains/solana/contracts/programs/ccip-router/src/pools.rs

```diff
@@ -46,14 +48,15 @@ pub fn calculate_token_pool_account_indices(
 
 pub struct TokenAccounts<'a> {
     pub user_token_account: &'a AccountInfo<'a>,
-    pub _token_billing_config: &'a AccountInfo<'a>,
+    pub token_billing_config: &'a AccountInfo<'a>,
     pub pool_chain_config: &'a AccountInfo<'a>,
     pub pool_program: &'a AccountInfo<'a>,
     pub pool_config: &'a AccountInfo<'a>,
     pub pool_token_account: &'a AccountInfo<'a>,
     pub pool_signer: &'a AccountInfo<'a>,
     pub token_program: &'a AccountInfo<'a>,
     pub mint: &'a AccountInfo<'a>,
+    pub fee_token_config: &'a AccountInfo<'a>,
```

Comment:
Hmm, maybe I'm confused about the semantics of BPS fees here. The reason why I felt like I needed this, is this snippet at `FeeQuoter.sol:669`:

```solidity
      // Only calculate bps fee if ratio is greater than 0. Ratio of 0 means no bps fee for a token.
      // Useful for when the FeeQuoter cannot return a valid price for the token.
      if (transferFeeConfig.deciBps > 0) {
        uint224 tokenPrice = 0;
        if (tokenAmount.token != feeToken) {
          tokenPrice = _getValidatedTokenPrice(tokenAmount.token);
        } else {
          tokenPrice = feeTokenPrice;
        }

        // Calculate token transfer value, then apply fee ratio
        // ratio represents multiples of 0.1bps, or 1e-5
        bpsFeeUSDWei = (tokenPrice._calcUSDValueFromTokenAmount(tokenAmount.amount) * transferFeeConfig.deciBps) / 1e5;
      }
```

It appears that the bps fee depends on the token price *of any token sent* (not just the fee token) because of the call to `_getValidatedTokenPrice`. Is there a way to access these prices that I'm not seeing, that would not require updating the lookup table?

Reply:
I've looked a bit more into this. In summary, what you did is correct in principle, you do need this account. However, your code has to be prepared for the possibility that it may not exist, in which case I suggest requiring the caller to pass in the zero address. Let me elaborate a bit:

Whenever bps fees are enabled, we do need the price for the token, as you mentioned, to understand how much extra to bill and how to convert that to the billing token (e.g. if we enable bps fees in ETH and the user pays with LINK, we need to do that conversion). However, we don't enable bps fees for most tokens.
Since ccip 1.5, we support users adding their own tokens to CCIP by having them create their own token pools in the source and destination chains. In those cases, we don't have any clue about the price of that token, so we couldn't enable bps fees even if we wanted to. 
For the few tokens that we might enable bps fees for _but not enable them as billing tokens used to pay for fees_ we can add the BillingTokenConfig as `enabled: false`. Current features already support this, so no change needed. We do need to ensure then that the offchain code will update that token's price from time to time in the commit report.

We can take a deeper look at another time on whether we want to merge TokenBilling and BillingTokenConfig, or at least split some of their properties in another way. But as far as this PR is concerned, what you did is correct, you just need to consider the case of a zero-address for the billingtokenconfig whenever bps are not enabled.